<?xml version="1.0"?>
<launch>
	<arg name="hw_or_sim" default="hw" />

	<group ns="frcrobot_rio">

		<!-- Load hardware interface -->
		<node name="frcrobot_rio_$(arg hw_or_sim)_interface" pkg="ros_control_boilerplate" type="frcrobot_$(arg hw_or_sim)_main"
			output="screen" launch-prefix="">
			<remap from="js0" to="joystick_states1"/>
		</node>

		<!-- Load controller settings -->
		<rosparam file="$(find ros_control_boilerplate)/config/talonfxpro_rio.yaml" command="load"/>
		<rosparam file="$(find ros_control_boilerplate)/config/robot_code_ready_controller_rio.yaml" command="load"/>

		<!-- Load controller manager -->
		<node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false"
			output="screen" args="spawn
				joint_state_controller 
				joint_state_listener_controller 
				robot_controller_state_controller 
				match_state_controller 
                rumble_controller
				rumble_state_controller
				robot_code_ready_controller"
			/>
			<!--
				talonfxpro_controller
				talonfxpro_state_controller
				talonfxpro_config_controller
				-->
	</group>
</launch>
